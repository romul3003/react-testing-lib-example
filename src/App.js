/* eslint-disable no-magic-numbers */
// import { useEffect, useState } from 'react'

// const App = () => {
//   const [data, setData] = useState(null)
//   const [toggle, setToggle] = useState(false)
//   const [value, setValue] = useState('')

//   const onClick = () => setToggle(prev => !prev)

//   useEffect(() => {
//     setTimeout(() => {
//       setData({})
//     }, 100)
//   }, [])

//   return (
//     <div className="container">
//       <h1 data-testId="value-elem">{value}</h1>
//       {toggle === true && (<div data-testId="toggle-elem">toggle</div>)}
//       {data && (<div style={{ color: 'red' }}>data</div>)}
//       <h1>Hello world</h1>
//       <button
//         type="button"
//         data-testId="toggle-button"
//         onClick={onClick}
//       >
//         Click me
//       </button>
//       <input
//         type="text"
//         placeholder="input value..."
//         onChange={event => setValue(event.target.value)}
//       />
//     </div>
//   )
// }

import Navbar from './components/Navbar'
import AppRouter from './router/AppRouter'

const App = () => (
  <div className="container" data-testid="app">
    <Navbar />
    <AppRouter />
  </div>
)

export default App
