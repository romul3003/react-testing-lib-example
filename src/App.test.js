import { fireEvent, render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import App from './App'

describe('TEST APP', () => {
  test('Should render <App />', () => {
    render(<App />)
    const helloWorldElem = screen.getByText(/hello world/i)
    const btn = screen.getByRole('button')
    const input = screen.getByPlaceholderText(/input value/i)

    expect(helloWorldElem).toBeInTheDocument()
    expect(btn).toBeInTheDocument()
    expect(input).toMatchSnapshot()
  })

  test('Should render text', async () => {
    render(<App />)
    // const helloWorldElem = screen.queryByText(/hello2/i)
    // expect(helloWorldElem).toBeNull()

    // screen.debug()
    const helloWorldElem = await screen.findByText(/data/i)
    expect(helloWorldElem).toBeInTheDocument()
    expect(helloWorldElem).toHaveStyle({ color: 'red' })
    // screen.debug()
  })

  test('Should click event', async () => {
    render(<App />)

    const btn = screen.getByTestId('toggle-button')
    expect(screen.queryByTestId('toggle-elem')).toBeNull()

    fireEvent.click(btn)
    expect(screen.getByTestId('toggle-elem')).toBeInTheDocument()

    fireEvent.click(btn)
    expect(screen.queryByTestId('toggle-elem')).toBeNull()
  })

  test('Should input event', async () => {
    render(<App />)

    const input = screen.getByPlaceholderText(/input value/i)
    expect(screen.queryByTestId('value-elem')).toContainHTML('')

    // synthetic event
    // fireEvent.input(input, {
    //   target: {
    //     value: '123123',
    //   },
    // })

    // close to user, handle events of keyboard inputs etc...
    userEvent.type(input, '123123')
    expect(screen.queryByTestId('value-elem')).toContainHTML('123123')
  })
})
