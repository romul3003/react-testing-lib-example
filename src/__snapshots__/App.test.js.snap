// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP Should render <App /> 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
